recipes/csbdeep/meta.yaml

```
@@ -0,0 +1,55 @@
+{% set name = "csbdeep" %}
+{% set version = "0.6.3" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/csbdeep-{{ version }}.tar.gz
+  sha256: be886751dde0d9e68dab49f2cc149f9bb38b718d6660c31d30722adbf7601327
+
+build:
+  entry_points:
+    - care_predict = csbdeep.scripts.care_predict:main
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
```

@uschmidt83 According to the docs this can work with TF2 or TF1. Which would be the minimum version for each? 
Something like this?

```suggestion
  run:
    - tensorflow >=2
```

---

tensorflow >= 1.4.1

See our legacy tests at https://github.com/CSBDeep/CSBDeep/blob/b0d2f5f344ebe65a9b4c3007f4567fe74268c813/.github/workflows/tests_legacy.yml#L52-L54

---

Initially I added the keras constraints (that were listed as extras) and the solver could not create the recipe @jaimergp

---

What if we add that `keras` req as `run_constrained` just in case, but depend on TF 2.*? Would that work?

---

Maybe 🤔 give it a go :)

---

I'm completely new to conda-forge and hadn't heard of `run_constrained` before.

[The docs](https://conda-forge.org/docs/maintainer/adding_pkgs.html#constraining-packages-at-runtime) indicate to me that `run_constrained` can be used to restrict `csbdeep` to `keras >=2.1.2,<2.4` if the user chooses to install it at all. Unfortunately, it's not that easy because `tensorflow` 2 didn't depend on the `keras` package for a little while (because they integrated it), then they changed their mind and it's now a required separate package that needs to be installed.

See https://github.com/tensorflow/tensorflow/releases/tag/v2.6.0 and https://github.com/tensorflow/tensorflow/releases/tag/v2.6.2 for more details.

---

If `tensorflow >=2` is required, then a suitable version of `keras` should be automatically installed due to the package requirements of `tensorflow`.